File: week 2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int calcLetters(string userinput);
int calcSentences(string userinput);
int calcWords(string userinput);

int main(void)
{
    string userinput = get_string("enter your paragraph: ");

    int Letter = calcLetters(userinput);
    int Sentence = calcSentences(userinput);
    int Word = calcWords(userinput);

    // printf("no. of letters: %d \n",Letter);
    // printf("no. of words: %d",Word);
    // printf("no. of Sentence: %d \n",Sentence);

    float l = ((float)Letter/(float)Word)*100;
    int s = ((float)Sentence/(float)Word)*100;
    float index = (0.0588 * l) - (0.296 * s) - 15.8;
    index = round(index);
    // printf("%d",(int)index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n",(int)index);
    }


}

int calcLetters(string userinput)
{
    int lenght = strlen(userinput);
    int letters = 0;
    for (int i = 0; i < lenght; i++)
    {
        if (isalpha(userinput[i]))
        {
            letters++;
        }
        else
        {
            continue;
        }

    }

    return letters;
}

int calcSentences(string userinput)
{
    int lenght = strlen(userinput);
    int sentences = 0;
    for (int i = 0; i < lenght; i++)
    {
        if (userinput[i] == '.' || userinput[i] == '!' || userinput[i] == '?')
        {
            sentences++;
        }
        else
        {
            continue;
        }
    }

    return sentences;
}

int calcWords(string userinput)
{
    int lenght = strlen(userinput);
    int words = 0;
    for (int i = 0; i <= lenght; i++)
    {
        if (userinput[i] == ' ' || userinput[i] == '\n' || userinput[i] == 0)
        {
            words++;
        }
        else
        {
            continue;
        }
    }
    return words;
}

File: week 6/sentimental-readability/readability.py
import cs50


def main():
    """Prompt user for input and print the result."""
    text = cs50.get_string("TEXT: ")

    letter = letters(text)
    sentence = sentences(text)
    word = words(text)

    L = (letter/word) *100
    S = (sentence/word) *100
    index = (0.0588 * L) - (0.296 * S) - 15.8

    if(index < 1):
        print("Before Grade 1")
    if(index > 1 and index <= 16):
        print(f"Grade {round(index)}")
    if(index > 16):
        print("Grade 16+")

    # print(index)
    # print(word)

def letters(text):
    counter = 0
    for i in text:
        if i.isalpha():
            counter+=1

    return counter

def sentences(text):
    counter = 0
    for i in text:
        if i == "." or i == "!" or i == "?":
            counter+=1
    return counter

def words(text):
    counter = 0
    for i in text:
        if i == " " or i == '\0':
            counter+=1
    if(text != None):
        counter+=1
    return counter
main()
